http4s/src/test/scala/io/isomarcte/errors4s/http4s/ResponseErrorProperties.scala

```diff
@@ -0,0 +1,144 @@
+            )
+          )
+        )
+
+        client
+          .expectOptionOr[String](testRequest)(ResponseError.fromResponseWithMaxSize(maxBodySize))
+          .attempt
+          .map(
+            _.fold(
+              e =>
+                e match {
+                  case e: ResponseError =>
+                    ((e.status ?= status) :| "Expected Status") &&
+                      ((e.headers ?= headers) :| "Expected Headers") &&
+                      ((e.errorBody ?= ResponseError.ErrorBody.FullErrorBody(errorBodyBytes)) :|
+                        "Expected ErrorBody") &&
+                      ((e.maxBodySize ?= maxBodySize) :| "Expected maxBodySize")
+                  case _ =>
+                    exception(e)
+                },
+              r => (falsified :| s"Got successful response, should have got ResponseError: ${r}")
+            )
+          )
+          .unsafeRunSync()
+    }
+
+  property("PartialErrorBody") =
+    forAllNoShrink(Arbitrary.arbitrary[Headers], invalidStatusGen, nonEmptyStringGen) {
+      (headers: Headers, status: Status, errorBody) =>
+        val errorBodyBytes: ByteVector = ByteVector.view(errorBody.getBytes(StandardCharsets.UTF_8))
+        val maxBodySize: Long          = errorBodyBytes.size - 1L
+        val client: Client[SyncIO] = Client[SyncIO](
+          Function.const(
+            Resource.pure[SyncIO, Response[SyncIO]](
+              Response(
+                status = status,
+                headers = headers,
+                body = Stream.chunk(Chunk.bytes(errorBodyBytes.toArray)).covary
+              )
+            )
+          )
+        )
+        val expectedBody: ByteVector =
+          if (errorBodyBytes.size <= 1) {
+            ByteVector.empty
+          } else {
+            errorBodyBytes.reverse.drop(1L).reverse
+          }
+
+        client
+          .expectOptionOr[String](testRequest)(ResponseError.fromResponseWithMaxSize(maxBodySize))
+          .attempt
+          .map(
+            _.fold(
+              e =>
+                e match {
+                  case e: ResponseError =>
+                    ((e.status ?= status) :| "Expected Status") &&
+                      ((e.headers ?= headers) :| "Expected Headers") &&
+                      ((e.errorBody ?= ResponseError.ErrorBody.PartialErrorBody(expectedBody)) :|
+                        "Expected ErrorBody") &&
+                      ((e.maxBodySize ?= maxBodySize) :| "Expected maxBodySize")
+                  case _ =>
+                    exception(e)
+                },
+              r => (falsified :| s"Got successful response, should have got ResponseError: ${r}")
+            )
+          )
+          .unsafeRunSync()
+    }
+
+  property("EmptyErrorBody") =
+    forAllNoShrink(Arbitrary.arbitrary[Headers], invalidStatusGen) { (headers: Headers, status: Status) =>
+      val errorBodyBytes: ByteVector = ByteVector.empty
+      val maxBodySize: Long          = errorBodyBytes.size - 1L
```

Comment:
WARN: Is `maxBodySize` allowed to be negative? What would the behavior be if there were a response body?

Reply:
@coacoas when the `maxBodySize` is negative, it is treated the same as if it was `0L`. This means you will always bet a `PartialErrorBody(ByteVector.empty)`. In the fs2 code, this is how the expression plays out.

 ```
scala> Stream.emits(List.empty[Int]).take(-1).toList
res0: List[Int] = List()
```

I'm fine with that personally. It's a clean way to encode a total function without dealing with some form of argument validation and keeping the API simple.